core/validators/signatures.go

```diff
@@ -106,6 +108,49 @@ type NodeIDAddress struct {
 	SubmitterAddress string
 }
 
+func (s *ERC20Signatures) OnTick(ctx context.Context, _ time.Time) {
```

Comment:
I don't think we need an `OnTick` we can just throw these calls in `func (t *Topology) BeginBlock()`. 

Actually looking at `OfferSignatures` is doesn't look like it checks that `v.signature == nil` before calling the callback, so I think we're always re-signed everything every block??

Reply:
I think you have a good point - it should check for nil. Although it doesn't regenerates forever - when the current signature is delivered from other nodes it will remove this ID from the list of retries.

Reply:
https://github.com/vegaprotocol/vega/blob/9409f972c7be326ad2eba695c2d47b1fddfb21c6/core/notary/notary.go#L185